### PR TITLE
need this pushed to dev

### PR DESCRIPTION
**Pull request by aditya7496**
develop was overridden with my reg and server files, I have copied samis code from both and checked that they work, all other files are unchanged please check if any other work was overridden and merge if not, else please add in and then merge.

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -57,33 +55,28 @@ server.get('/', async(req, res) => {
...
+    res.render("create.ejs", { name: req.user.firstName })
 })
 
 server.get('/login', checkNotAuthenticated, (req, res) => {
     res.render('login.ejs')
 })
 server.post('/login', checkNotAuthenticated, passport.authenticate("local", {
 
-    successRedirect: "/homepage",
+    successRedirect: "/create",
```
Comment (BoulterJackson) [inline src/server.js:66]: want to keep prev value "/homepage"

File: src/server.js [hunk 2]
```diff
@@ -124,8 +115,7 @@ server.post('/registration', checkNotAuthenticated,
             } catch (error) {
                 console.log(error)
                 req.flash("error", "User is already registered. Please login.");
-                res.redirect("/registration")//todo, results in a 302 status redirection code
-
+                res.redirect("/registration")
```
Comment (BoulterJackson) [inline src/server.js:118]: want to keep prev TODO comment

File: src/server.js [hunk 3]
```diff
@@ -164,8 +154,7 @@ server.post('/forgot-password', async(req, res) => {
                 id: user.id
             }
             const token = jwt.sign(payload, secret, { expiresIn: '15m' })
-            const resetLink = `localhost:8080/reset-password/${user.id}/${token}`//TODO: change local host to domain name
-
+            const resetLink = `localhost:8080/reset-password/${user.id}/${token}`
```
Comment (BoulterJackson) [inline src/server.js:157]: want to keep prev TODO comment

File: src/server.js [hunk 4]
```diff
@@ -254,11 +243,6 @@ server.post('/reset-password/:id/:token',
 
     })
 
-    server.get('/homepage', async(req, res) => {
-        res.render('homepage.ejs')
-    })
```
Comment (BoulterJackson) [inline src/server.js:259]: want to keep this function

File: views/registration.ejs [hunk 5]
```diff
@@ -36,9 +36,14 @@
...
-                      <div class="alert-danger" role="alert">
-                        <strong><%= messages.error %></strong>
+                      <% if ((validationErrors &&
+                      validationErrors.length)||messages.error) { %>
+                      <div class="alert alert-danger">
+                        <% if(messages.error) { %>
+                        <li><%= messages.error %></li>
+                        <% } %> <% validationErrors.forEach(error => { %>
+                        <li><%= error %></li>
+                        <% }) %>
```
Comment (BoulterJackson) [inline views/registration.ejs:46]: looks good

**Review by BoulterJackson — CHANGES_REQUESTED** [review]
looks good, just some small changes

File: src/server.js (hunk 1 above)
Comment (aditya7496) [reply src/server.js:66]: right forgot those changes were mine, thanks

File: src/server.js (hunk 2 above)
Comment (aditya7496) [reply src/server.js:118]: not sure whos comment this is but looks okay

File: src/server.js (hunk 3 above)
Comment (aditya7496) [reply src/server.js:157]: no comment needed

File: src/server.js (hunk 4 above)
Comment (aditya7496) [reply src/server.js:259]: looks good

**Pull request merged**